taskvine/src/worker/vine_worker.c

```diff
@@ -544,7 +544,12 @@ static void reap_process(struct vine_process *p, struct link *manager)
 	gpus_allocated -= p->task->resources_requested->gpus;
 
 	vine_gpus_free(p->task->task_id);
-	vine_sandbox_stageout(p, cache_manager, manager);
+
+	/* For a library process, do not stageout its sandbox until the watched logfile
```

Comment:
Do not add a special case for libraries.  If it has outputs, they should be staged out in the same way as a normal task.  (And he manager already knows how to deal with watched output files at stageout time.)